### PR TITLE
Issue/665

### LINKED ISSUE #665: Improved user searching
We should improve our user search that is shown when adding new affiliates, adding discount codes etc.

Currently when searching for a user via the affiliate discount fields, it will show all users, which means you can select the username of a pending affiliate.

Any other suggestions to improve the user search are welcome.

### PR DESCRIPTION
**Pull request by fjarrett**
Resolves #665 

This PR refactors the admin user search AJAX to be able to distinguish a specific set of affiliate users that should be included/excluded in query matches using a `data-affwp-status` input attribute.

**Some attribute value examples**

`none` - Only search users that **are not** affiliates (who have no affiliate status).
`any` - Only search users that are affiliates, with any status.
`active` - Only search users that are active affiliates.
`pending` - Only search users that are pending affiliates.

The only reserved values here are `none` and `any`, any custom status added in the future would be automatically supported.

Of course you can just omit this data attribute entirely and **all users** will be searched, affiliates and non-affiliates.

**Performance improvements**

I've also added a simple 500ms delay before AJAX requests are sent, rather than sending them immediately on the `keyup` event. This is a significantly more efficient way of achieving the same result.

### CODE REVIEW

**Comment by fjarrett** [comment]
@sumobi There was a random error in [this Travis job](https://travis-ci.org/affiliatewp/AffiliateWP/jobs/69166292), it just needs to be restarted.

**Comment by pippinsplugins** [comment]
Super awesome, can't wait to test this out!

File: includes/admin/ajax-actions.php [hunk 1]
```diff
@@ -3,37 +3,85 @@
...
+						'status' => $status,
+					)
+				);
+				$args = array( 'include' => array_map( 'absint', wp_list_pluck( $affiliates, 'user_id' ) ) );
+		}
+	}
+
+	$found_users = array_filter(
+		get_users( $args ),
+		function ( $user ) {
```
Comment (fjarrett) [inline includes/admin/ajax-actions.php]: Whoops, this requires PHP 5.3+, not sure what the compatibility rules are with AffiliateWP?

Comment (fjarrett) [inline includes/admin/ajax-actions.php]: Based on our Slack convo yesterday, it sounds like we can't do it this way.

I have a feeling this entire issue will be replaced by #754 anyway which I plan to start working on tomorrow and will incorporate some of the same principles/functionality introduced in this PR.

@pippinsplugins So maybe hold off on looking deeper into this one for now.

Comment (pippinsplugins) [inline includes/admin/ajax-actions.php]: Let's hold off for now.

**Comment by pippinsplugins** [comment]
After playing with this, I really, really like this and am going to go ahead and merge it.

We will bump the minimum version to 5.3.

**Pull request merged**